### ISSUE TITLE
Allow passing timer names to all libraries that create timers

### ISSUE THREAD

**Issue by rkoshak** [rkoshak/openhab-rules-tools#70]
This impacts:

- [ ] countdownTimer
- [ ] deferred
- [ ] gatekeeper
- [ ] loopingTimer
- [ ] timerMgr

Note, some of the above depend on timerMgr so that should be done first.

The behavior should be as follows:

- offer an options argument when creating a timer to provide a timer name
- if the name is not provided, use `ruleUID` or perhaps a combo of `ruleUID` and `key` in cases where a key is provided (e.g. timerMgr)
- if `ruleUID` isn't defined use `key`
- if 'ruleUID` and 'key` are undefined, use the name of the library (e.g. "TimerMgr")

I think it might be best to centralize the creation of the actual timers to a utility function so we don't have to reproduce the above logic over and over again.
